app/Leaf/Dynicon.php

```diff
@@ -31,112 +35,249 @@ class Dynicon
 
     public function __construct($file, $width)
     {
-        $this->file = $file;
-        $this->width = $width;
+        $this->cacheDir = realpath(self::CACHE_DIR);
+        $this->svgSourceDir = realpath(self::SVG_SOURCE_DIR);
 
-        if (!is_numeric($width) || $width <= 0) {
+        if ($this->cacheDir === false || $this->svgSourceDir === false) {
+            $this->outputError('Configuration error');

```

Comment:
Should we put a bit more information out to the user on what the configuration error is? Also for the end user what do they see? I think we would want to output a 500 error.